### PR TITLE
[test-only] delete connection tests

### PR DESCRIPTION
**Pull request by ScharfViktor**
related https://github.com/owncloud/web/issues/11688

- tests for deleting federated connection
- created bugs: #10213 #10216

### CODE REVIEW

File: tests/acceptance/TestHelpers/OcmHelper.php [hunk 1]
```diff
@@ -170,4 +170,38 @@ public static function listInvite(
...
+		string $user,
+		string $password,
+		string $userId,
+		string $idp
+	): ResponseInterface {
+		$url = self::getFullUrl($baseUrl, 'delete-accepted-user');
+		$body = [
+			  "idp" => $idp,
+			  "user_id" => $userId
+			];
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/OcmHelper.php:197]:
```suggestion
		$body = [
			"idp" => $idp,
			"user_id" => $userId
		];
```

File: tests/acceptance/bootstrap/OcmContext.php [hunk 2]
```diff
@@ -230,6 +230,25 @@ public function userFindsAcceptedUsers(string $user): void {
...
 
+	/**
+	 *
+	 * @param string $user
+	 * @param string $ocmUserName
+	 *
+	 * @return array
+	 * @throws GuzzleException
+	 */
+	public function getAcceptedUserByName(string $user, $ocmUserName): array {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcmContext.php:241]:
```suggestion
	public function getAcceptedUserByName(string $user, string $ocmUserName): array {
```

File: tests/acceptance/bootstrap/OcmContext.php [hunk 3]
```diff
@@ -230,6 +230,25 @@ public function userFindsAcceptedUsers(string $user): void {
...
+		$this->userFindsAcceptedUsers($user);
+		$responseArray = ($this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse()));
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcmContext.php:243]: It's a helper function so it's better to not save the response here
```suggestion
		$response = $this->findAcceptedUsers($user);
		$responseArray = ($this->featureContext->getJsonDecodedResponse($response));
```

File: tests/acceptance/features/apiOcm/deleteFederatedConnections.feature [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            "type": "array",
+            "minItems": 0,
+            "maxItems": 0
+          }
+        }
+      }
+      """
+
+  @issue-10216
+  Scenario: user cannot find the federated user if the user has deleted the connection
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/deleteFederatedConnections.feature:39]: IMO, this makes scenario more clear
```suggestion
  Scenario: local user should not be able to find federated user after federated user has deleted connection
```

File: tests/acceptance/features/apiOcm/deleteFederatedConnections.feature [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            "type": "array",
+            "minItems": 0,
+            "maxItems": 0
+          }
+        }
+      }
+      """
+
+  @issue-10216
+  Scenario: user cannot find share if he has deleted the connection
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/deleteFederatedConnections.feature:66]:
```suggestion
  Scenario: federated user should not be able to find federated share after federated user has deleted connection
```

File: tests/acceptance/features/apiOcm/deleteFederatedConnections.feature [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  As a user
+  I want to delete federated connections if they are no longer needed
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And using server "REMOTE"
+    And user "Brian" has been created with default attributes and without skeleton files
+
+
+  Scenario: users deletes federated connection
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/deleteFederatedConnections.feature:12]:
```suggestion
  Scenario: federated user deletes the federated connection
```

File: tests/acceptance/features/apiOcm/deleteFederatedConnections.feature [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            "type": "array",
+            "minItems": 0,
+            "maxItems": 0,
+          }
+        }
+      }
+      """
+
+  @issue-10213
+  Scenario: user cannot find share if the user has deleted the connection
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/deleteFederatedConnections.feature:103]:
```suggestion
  Scenario: federated user should not be able to find federated share after local user has deleted connection
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 8]
```diff
@@ -91,6 +95,93 @@ Feature: an user shares resources usin ScienceMesh application
       }
       """
 
+
+  Scenario: users shares file to federation users after receiver accepted invitation
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:99]:
```suggestion
  Scenario: local user shares a file to federated user
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 9]
```diff
@@ -171,3 +266,88 @@ Feature: an user shares resources usin ScienceMesh application
         }
       }
       """
+
+
+  Scenario: users shares file to federation users after accepting invitation
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:271]:
```suggestion
  Scenario: federated user shares a file to local user
```

File: tests/acceptance/features/apiOcm/share.feature [hunk 10]
```diff
@@ -171,3 +266,88 @@ Feature: an user shares resources usin ScienceMesh application
...
+    Given using server "LOCAL"
```
Comment (saw-jan) [inline tests/acceptance/features/apiOcm/share.feature:272]: maybw group this with above similar scenario

Comment (ScharfViktor) [reply tests/acceptance/features/apiOcm/share.feature:272]: done

File: tests/acceptance/bootstrap/OcmContext.php [hunk 11]
```diff
@@ -230,6 +230,25 @@ public function userFindsAcceptedUsers(string $user): void {
...
+	 * @param string $user
+	 * @param string $ocmUserName
+	 *
+	 * @return array
+	 * @throws GuzzleException
+	 */
+	public function getAcceptedUserByName(string $user, string $ocmUserName): array {
+		$response = $this->findAcceptedUsers($user);
+		$responseArray = ($this->featureContext->getJsonDecodedResponse($response));
+		foreach ($responseArray as $findUser) {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcmContext.php:244]:
```suggestion
		$users = ($this->featureContext->getJsonDecodedResponse($response));
		foreach ($users as $user) {
```

File: tests/acceptance/bootstrap/OcmContext.php [hunk 12]
```diff
@@ -230,6 +230,25 @@ public function userFindsAcceptedUsers(string $user): void {
...
+			if (strpos($findUser["display_name"], $ocmUserName) !== false) {
+				return $findUser;
+			}
+		}
+		throw new \Exception("User with name '{$ocmUserName}' not found in the accepted users.");
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcmContext.php:249]:
```suggestion
		throw new \Exception("Could not find user with name '{$ocmUserName}' in the accepted users list.");
```

File: tests/acceptance/bootstrap/OcmContext.php [hunk 13]
```diff
@@ -268,4 +287,51 @@ public function userListsCreatedInvitations(string $user): void {
...
+	 *
+	 * @param string $user
+	 * @param string $ocmUser
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userHasDeletedFederatedConnectionWithUser(string $user, string $ocmUser): void {
+		$response = $this->deleteConnection($user, $ocmUser);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200, "failed while deliting connection with user $ocmUser", $response);
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/OcmContext.php:315]:
```suggestion
		$this->featureContext->theHTTPStatusCodeShouldBe(200, "failed while deleting connection with user $ocmUser", $response);
```

**Review by saw-jan — COMMENTED** [review]
others look good

**Review by saw-jan — APPROVED** [review]

**Pull request merged**